api/src/main/java/org/openmrs/module/legacyui/GeneralUtils.java

```diff
@@ -68,4 +83,69 @@ public static Concept getConcept(String id) {
 		
 		return cpt;
 	}
+	
+	/**
+	 * @see OrderExtensionService#getProviderForUser(User)
+	 */
+	@Transactional(readOnly = true)
+	public static Provider getProviderForUser(User user) {
+		ProviderService ps = Context.getProviderService();
+		Collection<Provider> providers = ps.getProvidersByPerson(user.getPerson(), true);
+		if (providers.isEmpty()) {
+			throw new IllegalStateException("User " + user + " has no provider accounts, unable to create orders");
+		}
+		return providers.iterator().next();
+	}
+	
+	/**
+	 * Copied from OpenMRS core 1.9.x
+	 * See https://github.com/openmrs/openmrs-core/blob/1.9.x/api/src/main/java/org/openmrs/order/OrderUtil.java#L52
+	 * 
+	 * Discontinues all current orders for the given <code>patient</code>
+	 * 
+	 * @param patient
+	 * @param discontinueReason
+	 * @param discontinueDate
+	 * @see OrderService#discontinueOrder(org.openmrs.Order, Concept, Date)
+	 * @should discontinue all orders for the given patient if none are yet discontinued
+	 * @should not affect orders that were already discontinued on the specified date
+	 * @should not affect orders that end before the specified date
+	 * @should not affect orders that start after the specified date
+	 */
+	public static void discontinueAllDrugOrders(Patient patient, Concept discontinueReason, Date discontinueDate) {
```

Comment:
It sounds like this was previously in an OrderUtil method, but really should be in a Transactional service method, as it is discontinuing multiple orders in a loop and we want that entire operation to be transactional.  Can you move this into the service?